www/src/templates/template-docs-markdown.js

```diff
@@ -15,9 +15,26 @@ import docsHierarchy from "../data/sidebars/doc-links.yaml"
 // Find the guides in the sidebar YAML.
 const guides = docsHierarchy.find(group => group.title === `Guides`).items
 
+// Search through the guides and their children to find the first item that
+// matches the given slug
+const findBySlug = (guides, slug) => {
```

Comment:
This looks better, but I wonder if we could make it less side-effecty with a `.find` and perhaps a flatten? Here's a start:

```js
const flatten = arr => arr.reduce((merged, item) => merged.concat(Array.isArray(item.items) ? flatten(item.items) : item), [])

const findBySlug = (guides, slug) => {
  return flatten(guides)
    .find(item => item.link === slug)
}
```

Reply:
Presuming we have a structure like (for guides):

```js
const guides = [
    {
      link: '/test'
    },
    {
      items: [
        {
          link: '/test2'
        }
      ]
    }
  ]
```

this seems to work just fine!

Reply:
I had a little play around with this but I don't think flatten is right here, as we want to grab the child items from the item with the matching slug. I rejigged some other bits while playing around with this though, thank you 👍